Rpackage/R/growtree.R

```diff
@@ -0,0 +1,72 @@
+growtree <- structure(function(target.mat, feature.mat, depth=0, maxdepth = Inf,
+                               margin=0.0, loss="hinge", id = 1L, min_sample = 1,
+                               pred=NULL, side = NULL) {
+
+  #if depth exceeds we stop the further growth of tree
+  if (depth >= maxdepth) return(partynode(id = id, info = pred[[side]]))
+  
+
+  #if sample per node is less than minimum, we stop tree growth
+  if(length(target.mat[,1]) <= min_sample) return(partynode(id = id, info = pred[[side]]))
+
+  #split the tree at the node.
+  #print(length(feature.mat[,2]))
+  sp <- bestsplit(target.mat, feature.mat, margin, loss)
+  splt <- sp
+
+  #if no split, we stop
+  if (is.null(sp)) return(partynode(id = id, info = pred[[side]]))
+
+  #partysplit object
+  sp <- partysplit(varid = sp$varid, breaks = sp$br)
+
+  #the kids left and right defined
+  kid <- NULL
+  kid$left <- NULL
+  kid$right <- NULL
+  tar <- NULL
+  tar$left <- NULL
+  tar$right <- NULL
+
+  #splitting into kids
+  kidids <- kidids_split(sp, data = feature.mat) #list of right left kids
+  #print(length(kidids))
```

Comment:
Don't forget to delete commented prints before we merge :)

Reply:
done